x-pack/plugins/security_solution/public/app/home/template_wrapper/index.tsx

```diff
@@ -73,11 +77,17 @@ export const SecuritySolutionTemplateWrapper: React.FC<SecuritySolutionPageWrapp
     const { show: isShowingTimelineOverlay } = useDeepEqualSelector((state) =>
       getTimelineShowStatus(state, TimelineId.active)
     );
+    const endpointMetadataIndex = useMemo<string[]>(() => {
+      return [ENDPOINT_METADATA_INDEX];
+    }, []);
+    const [, { indexExists: metadataIndexExists }] = useFetchIndex(endpointMetadataIndex, true);
+    const { indicesExist } = useSourcererScope();
+    const securityIndicesExist = indicesExist || metadataIndexExists;
 
     /* StyledKibanaPageTemplate is a styled EuiPageTemplate. Security solution currently passes the header and page content as the children of StyledKibanaPageTemplate, as opposed to using the pageHeader prop, which may account for any style discrepancies, such as the bottom border not extending the full width of the page, between EuiPageTemplate and the security solution pages.
      */
 
-    return (
+    return securityIndicesExist ? (
```

Comment:
🤔  I'm wondering if this is the right place for this change to occur, which impact all areas of security solution.

For endpoint, one might (maybe?) want to start configuring policies (ex. trusted applications, event filters, etc.) before any data is available in the Endpoint metadata index. This change makes it impossible for the user to do that until at least one of the checked indexes has data in it.

Do I have that right?

I'm thinking that each section of security solution should problaby have its won way to determine if the onboarding page should be displayed or not.

cc/ @kevinlog

Reply:
In general I agree, but for now we just need **something** up. This suggestion would require a lot more coordination that I"m not sure we have time for in 7.16.